shared/go/aws/cloudwatchlogs/cloudwatch.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright (C) 2022 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloudwatchlogs
+
+import (
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudwatchlogs"
+	"github.com/falcosecurity/plugin-sdk-go/pkg/sdk"
+	"github.com/falcosecurity/plugin-sdk-go/pkg/sdk/plugins/source"
+	"github.com/falcosecurity/plugins/shared/go/aws/session"
+)
+
+// Client represents a client for CloudwatchLogs API
+type Client struct {
+	*cloudwatchlogs.CloudWatchLogs
+}
+
+// CreateClient returns a Client for CloudwatchLogs API
+func CreateClient(cfgs *aws.Config) *Client {
+	return &Client{
+		CloudWatchLogs: cloudwatchlogs.New(session.CreateSession(), cfgs),
+	}
+}
+
+// CreateFilter returns a filter for the query to CloudwatchLogs API
+func CreateFilter(startTime, endTime int64, filterPattern, logGroupName, logStreamNamePrefix, logStreamNames string) *cloudwatchlogs.FilterLogEventsInput {
+	return &cloudwatchlogs.FilterLogEventsInput{
+		StartTime:           time.Now().Add(-10 * time.Second).UnixMilli(),
+		FilterPattern:       aws.String(filterPattern),
+		LogGroupName:        aws.String(logGroupName),
+		LogStreamNamePrefix: aws.String(logStreamNamePrefix),
+		LogStreamNames:      aws.String(logStreamNames),
+	}
+}
+
+// Open returns an instance with the functionn called to retrieve logs
+func (client *Client) Open(filters *cloudwatchlogs.FilterLogEventsInput, refresh int64) (source.Instance, error) {
```

Comment:
A downside of creating a `source.Instance` internally is that developers have _no control_ over the data written on each event, which would always be the raw content of each CW log entry. For example, this can be limiting for cases like a future AKS integration, in which we may want to pre-process the log entries before writing them in the `k8saudit` event source.

What about passing an optional `func([]byte) []byte` transformation function? Alternatively, this could return a read-only `source.PushEvent` chan, so that users can choose to preprocess each event before pushing it to the event source.

Reply:
I think the easiest way is to return a `chan []byte`, wdyt?

Reply:
Yeah I agree. `source.PushEvent` can contain either a `[]byte` or a `error`, so it can be handier to propagate errors in the channel.